fetools-app/src/components/ColorPicker/CustomPicker.jsx

```diff
@@ -0,0 +1,135 @@
+  const intervalMouseClickRef = useRef(null);
+
+  const [mouseCoor, setMouseCoor] = useState({x:0,y:0})
+  const [currentColor, setCurrentColor] = useState(getHexString(colorData.color))
+
+  useEffect(()=>{
+    markerRef.current.children[0].style.background = colorData.color
+    createCanvasGradients()
+    canvasRef.current.currentColor = currentColor
+
+    return ()=>{
+      stopInterval(intervalMouseMoveRef)
+    }
+
+  },[mouseCoor])
+
+  return(
+    <>
+    <div>
+      <div className="relative">
+        <canvas 
+        ref={canvasRef} 
+        id="color-picker"
+        onMouseMove={(e)=>startInterval(e,handleOnMouseMove,intervalMouseMoveRef)}
+        onClick={(e)=>handleClick(e,true)}
+        onMouseDown={(e)=>startInterval(e,handleClick,intervalMouseClickRef)}
+        onMouseUp={()=>(stopInterval(intervalMouseClickRef), stopInterval(intervalMouseMoveRef))}
+        onMouseLeave={()=>(stopInterval(intervalMouseClickRef), stopInterval(intervalMouseMoveRef))}
+        className="relative z-0">
+        </canvas>
+        <div ref={markerRef} 
+        className={`
+        marker absolute leading-none rounded-full border-4 outline outline-1 outline-slate-600  w-5 h-5 mt-[-11px] ml-[-8px] pointer-events-none`}>
+          {createPickerMarker()}
+        </div> 
+      </div>
+
+    </div>  
+    </>
+  )
+
+    function createCanvasGradients(){
+      const ColorCtx = canvasRef.current.getContext('2d');  // This create a 2D context for the canvas
+
+      const hue = colorData.hue;
+      //Make HueGradient
+      const gradientH = ColorCtx.createLinearGradient(0, 0, ColorCtx.canvas.width, 0);
+      gradientH.addColorStop(0, '#fff');
+      gradientH.addColorStop(1, hue);
+      ColorCtx.fillStyle = gradientH;
+      ColorCtx.fillRect(0, 0, ColorCtx.canvas.width, ColorCtx.canvas.height);
+
+      // Create a Vertical Gradient(white to black)
+      const gradientV = ColorCtx.createLinearGradient(0, 0, 0, ColorCtx.canvas.height);
+      gradientV.addColorStop(0, 'rgba(0,0,0,0)');
+      gradientV.addColorStop(1, '#000');
+      ColorCtx .fillStyle = gradientV;
+      ColorCtx .fillRect(0, 0, ColorCtx.canvas.width, 
+      ColorCtx .canvas.height); 
+
+    }
+
+    function createPickerMarker(){
+      return(
+        <div className={`rounded-full border border-slate-300 w-full h-full`}></div>
+      )
+    }
+
+    function handleClick(e, firstClick){
+      colorData.color = canvasRef.current.currentColor
+      handleColorChange({...colorData})
+      calculateMarkerPosition()
+      if(!firstClick){
+        canvasRef.current.click()
+      }
+      
+    }
+
+    function startInterval(e, func, ref){
+      if (ref.current) return;
+      ref.current = setInterval(() => {
+        func(e)
+      }, 50);
+    }
+
+    function stopInterval(ref){
+      if (ref.current) {
+        clearInterval(ref.current);
+        ref.current = null;
+      }
+    }
+
+    function handleOnMouseMove(e){
```

Comment:
The tracker currently feels a little sluggish; if there is anything we can do to make it more responsive, that would be great. This is not a criticism; the logic is clean and makes sense, and you did a great job here.